flytekit/core/pod_template.py

```diff
@@ -0,0 +1,22 @@
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+from kubernetes.client.models import V1PodSpec
+
+from flytekit.exceptions import user as _user_exceptions
+
+PRIMARY_CONTAINER_DEFAULT_NAME = "primary"
+
+
+@dataclass
+class PodTemplate(object):
+    pod_spec: V1PodSpec
+    primary_container_name: str = PRIMARY_CONTAINER_DEFAULT_NAME
+    labels: Optional[Dict[str, str]] = None
+    annotations: Optional[Dict[str, str]] = None
+
+    def __post_init__(self):
+        if not self.pod_spec:
+            raise _user_exceptions.FlyteValidationException("A pod spec cannot be undefined")
```

Comment:
Does this need to be required? If a user doesn't provide it, we can just generate an empty one right? It would make the API more ergonomic IMO.

Reply:
I copied this from k8s_pod task, but I think your statement makes sense. The empty pod spec will be filled with default values in `_serialize_pod_spec`

Reply:
drop it

Reply:
addressed in the new commits